compiler/src/iree/compiler/Codegen/LLVMCPU/LLVMCPULowerToUKernels.cpp

```diff
@@ -152,6 +152,176 @@ static FailureOr<IREE::Codegen::UKernelOpInterface> matchDAGForUKernel(
       genericMicroKernelOp.getOperation());
 }
 
+static FailureOr<IREE::Codegen::UKernelOpInterface> matchDAGForUKernel(
+    RewriterBase &rewriter, tensor::PackOp op) {
+  if (!op.hasTensorSemantics()) {
+    return rewriter.notifyMatchFailure(
+        op, "operations needs to have tensor semantics");
+  }
+  Value in = op.getSource();
+  Value out = op.getDest();
+  auto inType = in.getType().cast<ShapedType>();
+  auto outType = out.getType().cast<ShapedType>();
```

Comment:
[optional] I don't have a strong opinion. Just an FYI that we can use `op.getSourceType()` and `op.getDestType()`.

Reply:
Here as I also use the values `in` and `out`, I like to keep `inType` and `outType` defined from them so that they are known, by construction, to actually be their type.